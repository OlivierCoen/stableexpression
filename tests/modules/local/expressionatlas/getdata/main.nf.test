nextflow_process {

    name "Test Process EXPRESSIONATLAS_GETDATA"
    script "modules/local/expressionatlas/getdata/main.nf"
    process "EXPRESSIONATLAS_GETDATA"
    tag "getdata"
    tag "module"

    test("Transcriptome Analysis of the potato (rnaseq)") {

        tag "getdata_potato"

        when {

            process {
                """
                input[0] = "E-MTAB-552"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)") {

        tag "getdata_arabido"

        when {

            process {
                """
                input[0] = "E-TABM-1007"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Arabidopsis Geo dataset") {

        tag "getdata_arabido_geo"

        when {

            process {
                """
                input[0] = "E-GEOD-62537"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("E-MTAB-5132") {

        tag "E-MTAB-5132"

        when {

            process {
                """
                input[0] = "E-MTAB-5132"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
