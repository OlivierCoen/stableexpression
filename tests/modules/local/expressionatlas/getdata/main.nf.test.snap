{
    "Transcriptome Analysis of the potato": {
        "content": [
            {
                "0": [
                    "E-MTAB-552.rnaseq.design.csv:md5,142d6bb20d784ba414fa76a6e73bb37f"
                ],
                "1": [
                    "E-MTAB-552.rnaseq.raw.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "metadata": [
                    "E-MTAB-552.rnaseq.design.csv:md5,142d6bb20d784ba414fa76a6e73bb37f"
                ],
                "normalised": [
                    
                ],
                "raw": [
                    "E-MTAB-552.rnaseq.raw.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T13:31:14.414421877"
    },
    "arabidopsis issue": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        "E-GEOD-62537",
                        "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                    ]
                ],
                "2": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "design": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "normalised": [
                    [
                        "E-GEOD-62537",
                        "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                    ]
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-03T19:19:10.505238238"
    },
    "Transcriptome Analysis of the potato (rnaseq)": {
        "content": [
            {
                "0": [
                    "E_MTAB_552_rnaseq.design.csv:md5,b81490696f638e90c1cf14236bb0c08c"
                ],
                "1": [
                    "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_MTAB_552_rnaseq.design.csv:md5,b81490696f638e90c1cf14236bb0c08c"
                ],
                "normalised": [
                    
                ],
                "raw": [
                    "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:40:48.88652487"
    },
    "Arabidopsis Geo dataset": {
        "content": [
            {
                "0": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "1": [
                    
                ],
                "2": [
                    "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "normalised": [
                    "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:41:28.635715975"
    },
    "E-MTAB-5132": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "normalised": [
                    
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-01T18:55:07.888658728"
    },
    "Transcription profiling by array of Arabidopsis mutant for fis2": {
        "content": [
            {
                "0": [
                    "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                ],
                "1": [
                    
                ],
                "2": [
                    "E-TABM-1007.A-AFFY-2.normalised.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "metadata": [
                    "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                ],
                "normalised": [
                    "E-TABM-1007.A-AFFY-2.normalised.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T13:28:23.604650949"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "E-MTAB-552_rnaseq.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ],
                "csv": [
                    "E-MTAB-552_rnaseq.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T18:28:01.099092539"
    },
    "Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)": {
        "content": [
            {
                "0": [
                    "E_TABM_1007_A_AFFY_2.design.csv:md5,120f63cae2193b97d7451483bdbbaab1"
                ],
                "1": [
                    
                ],
                "2": [
                    "E_TABM_1007_A_AFFY_2.normalised.csv:md5,a3afe33d7eaed3339da9109bf25bb3ed"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_TABM_1007_A_AFFY_2.design.csv:md5,120f63cae2193b97d7451483bdbbaab1"
                ],
                "normalised": [
                    "E_TABM_1007_A_AFFY_2.normalised.csv:md5,a3afe33d7eaed3339da9109bf25bb3ed"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:41:08.294214693"
    }
}